tests/purs/failing/NestedRecordTypeError.out

```diff
@@ -0,0 +1,41 @@
+Error found:
+in module [33mA[0m
+at tests/purs/failing/NestedRecordTypeError.purs:6:3 - 6:34 (line 6, column 3 - line 6, column 34)
+
+  Could not match type
+  [33m                   [0m
+  [33m  ( c :: { d :: Int[0m
+  [33m         }         [0m
+  [33m  ...              [0m
+  [33m  )                [0m
+  [33m                   [0m
+  with type
+  [33m                  [0m
+  [33m  ( b :: { z :: t0[0m
+  [33m         | t1     [0m
+  [33m         }        [0m
+  [33m  ...             [0m
+  [33m  | t2            [0m
+  [33m  )               [0m
+  [33m                  [0m
```

Comment:
Shouldn't there be a `g` mentioned somewhere in here? What's up with that?

Reply:
Both yes and no. Since the unification is done on each field we are reporting on a record field. In a perfect world I would like it to comment `while matching row 'a'`.

That said, I would expect a second error complaining about `g`. So this definitely warrants further investigation.

Reply:
I looked into this. It's caused by not catching any possible errors thrown from `parU matches id` **and** `uncurry unifyTails rest`. Because `a` is a label shared between the two rows, `parU matches id` will recurse down on the `a` label. And because the rows under the label `a` do not unify, an error is thrown. Because the error short-circuits this computation, the `uncurry unifyTails rest` code is never evaluated. Thus, the second error does not appear.

Reply:
This issue has been resolved.